### PR TITLE
Part5

### PR DESCRIPTION
**Pull request by 0bi0n3**
Added my new functions under `// new function` comments.

In `int main` for the console printing, I placed the function calls under `//4` after the previous function calls.

I also tried experimenting with `rand() % 2` which allowed me to do some random number generating. Please let me know if you'd like this removed! :)

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -191,8 +248,39 @@ struct MidiKeyboard
...
+                std::cout << "   PROGRAMMING      **\n";
+                std::cout << "   FOR                   **\n";
+                std::cout << "   MUSICIANS                  **\n";
+                std::cout << "   IS                    **\n";
+                std::cout << "   GREAT!           **\n";
+                std::cout << "               **\n";
+                std::cout << "          **\n";
+                std::cout << "     **\n";
+                std::cout << "**\n";
+            }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -237,13 +325,71 @@ struct RolandTB303
...
+    if (X == 'A')
+    {
+    std::cout << "Loading previous sequence... " << X << "\n";
+    int sequenceMemory = 25;
+    std::cout << "Sequence step total: " << sequenceMemory << "\n";
+    for (int seq = 0; seq < 25; ++seq)
+        {
+            std::cout << "Step " << (seq + 1) << " loaded\n";
+        }
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -237,13 +325,71 @@ struct RolandTB303
...
+    int totalMidiCount = midiNotesIn;
+    for (int note = 0; note < midiNotesIn; ++note)
+        {
+            std::cout << "Syncing MIDI...\n";
+            if (midiNotesIn != 0)
+            {
+                std::cout << "MIDI sync progress " << (note + 1) << " out of " << midiNotesIn << " \n" << "MIDI notes remaining before full sync " << (totalMidiCount -= 1) << " \n";
+            }
+        }
+    return totalMidiCount;
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -311,8 +457,25 @@ struct Dictaphone
...
+int Dictaphone::flagNumberOfSpeakers()
+{    
+    int Voices = 4;
+    for (int headCounter = 0; headCounter < Voices; ++headCounter)
+        {
+            adjustMicrophoneSensitivity /= 5.0;
+            std::cout << "Number of voices present " << Voices << " \n";
+            std::cout << "Microphone sensitiy being actively reduced by " << (adjustMicrophoneSensitivity * 5) << " \n";
+        }
+
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -311,8 +457,25 @@ struct Dictaphone
...
     void recordingOn(bool activeRecording);
     std::string displayAvailableSpace(Dictaphone dictaphoneStorage);
+
+    // new function
+    int flagNumberOfSpeakers();
 };
 
+int Dictaphone::flagNumberOfSpeakers()
+{    
+    int Voices = 4;
```
Comment (matkatmusic) [inline main.cpp]: Fixme: variable name does not conform to course coding standard
- [x] revise

File: main.cpp [hunk 6]
```diff
@@ -396,8 +593,27 @@ struct CPU
...
+    std::cout << "Increasing RAM allocation...\n";
+    for (int i = 0; i < ramSize; ++i)
+        {
+            ramMemAlloc += 2;
+            std::cout << "RAM allocation increased to " << ramMemAlloc << " Gbs\n";
+            if (ramMemAlloc != (ramSize * 2))
+            {
+            ++i;
+            }
+        }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 7]
```diff
@@ -432,8 +648,29 @@ struct GPU
...
+                changeRenderRate = 0;
+                std::cout << "Render rate at minimum! Please increase rate.\n";
+            }
+            else
+            {
+                changeRenderRate /= 2;
+                std::cout << "Rendering decreased by a factor of " << (changeRenderRate * 10) << " Hz\n";
+                changeRenderRate = 0;
+            }
+        }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 8]
```diff
@@ -485,8 +722,30 @@ struct Memory
...
+            {
+                bufferAllocate += 256;
+                std::cout << "Buffer block of " << bufferAllocate << " allocated\n";
+                ++buffer;
+            }
+            else
+            {
+                std::cout << "Buffer request too small\n";
+            }
+        }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 9]
```diff
@@ -527,8 +787,24 @@ struct UsbPort
...
+    
+    for(int flashCount = 0; flashCount < connectedDevices; ++flashCount)
+        {
+            int totalDevices = flashCount;
+            if (connectedDevices != totalDevices)
+            {
+                std::cout << "Flashing connection LED for device: " << (totalDevices + 1) << " \n";
+                
+            }
+        }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 10]
```diff
@@ -578,9 +856,35 @@ struct GamingLaptop
...
+            {
+                isConnected = true;
+                std::cout << "Successfully connected to multiplayer server on attempt " << (connectionAttempts + 1) << ".\n";
+                break;
+            }
+            else
+            {
+                std::cout << "Attempt " << (connectionAttempts + 1) << " to server failed. Attempting to reconnect...\n";
+            }
+        }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 5 above)
Comment (0bi0n3) [reply main.cpp]: Ah yes! That slipped through

File: main.cpp [hunk 11]
```diff
@@ -237,13 +329,76 @@ struct RolandTB303
...
+    {
+    std::cout << "Loading previous sequence... " << X << "\n";
+    int sequenceMemory = 25;
+    std::cout << "Sequence step total: " << sequenceMemory << "\n";
+        
+    for (int seq = 0; seq < 25; ++seq)
+    {
+        std::cout << "Step " << (seq + 1) << " loaded\n";
+    }
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Always proofread your Pull Request before sending it to me for review.

File: main.cpp (hunk 11 above)
Comment (0bi0n3) [reply main.cpp]: Thanks, I do attempt to fully-proof read before sending. I did find another and fixed that too.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**